src/server/qgswfsserver.cpp

```diff
@@ -2066,6 +2071,12 @@ QDomElement QgsWFSServer::createFeatureGML2( QgsFeature* feat, QDomDocument& doc
   featureElement.appendChild( typeNameElement );
 
   const QgsGeometry* geom = feat->constGeometry();
+  if ( isMultiGeom && QgsWKBTypes::isSingleType( QGis::fromOldWkbType( geom->wkbType() ) ) )
+  {
+    QgsGeometry* cloneGeom = new QgsGeometry( *geom );
+    cloneGeom->convertToMultiType();
+    geom = cloneGeom;
```

Comment:
I think you have a memory leak in this case because the new geometry is never deleted.

Reply:
This shouldn't be using a geometry pointer in any case -- it should be using QgsGeometry values instead. They are implicitly shared so inexpensive to copy.